src/include/common/file_utils.h

```diff
@@ -15,6 +15,24 @@
 namespace kuzu {
 namespace common {
 
+enum class FileLockType : uint8_t { NO_LOCK = 0, READ_LOCK = 1, WRITE_LOCK = 2 };
+
+class FileFlags {
```

Comment:
This class should be cleaned after you merge locking logic into existing `openFile` function.